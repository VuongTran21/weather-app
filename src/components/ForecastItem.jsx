import { Card, Col } from 'react-bootstrap';
import { getDate } from '../utils';
import PropTypes from 'prop-types';

export default function ForecastItem({ forecast }) {
  const imageUrl = `https://www.metaweather.com/static/img/weather/png/${forecast.weather_state_abbr}.png`;

  return (
    <Col key={forecast.id}>
      <Card>
        <Card.Img className="card-image" variant="top" src={imageUrl} />
        <Card.Body>
          <Card.Title>{getDate(forecast.applicable_date)}</Card.Title>
          <Card.Text>Min temp: {Math.floor(forecast.min_temp)}</Card.Text>
          <Card.Text>Max temp: {Math.floor(forecast.max_temp)}</Card.Text>
        </Card.Body>
      </Card>
    </Col>
  );
}

ForecastItem.propTypes = {
  forecast: PropTypes.shape({
    id: PropTypes.number.isRequired,
    max_temp: PropTypes.number.isRequired,
    min_temp: PropTypes.number.isRequired,
    applicable_date: PropTypes.string.isRequired,
    weather_state_abbr: PropTypes.string.isRequired,
  }),
};
