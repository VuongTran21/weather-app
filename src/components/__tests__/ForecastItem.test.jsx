import React from 'react';
import { render } from '@testing-library/react';

import ForecastItem from '../ForecastItem';

describe('ForecastItem Component', () => {
  it('should render forecast item', () => {
    const props = {
      id: 1,
      weather_state_abbr: 'icon',
      applicable_date: '2021-03-15',
      min_temp: 18,
      max_temp: 20
    };

    const { container } = render(<ForecastItem forecast={props} />);
    expect(container).toMatchSnapshot();
  });
});
