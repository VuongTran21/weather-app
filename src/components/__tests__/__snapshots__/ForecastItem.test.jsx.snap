// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem Component should render forecast item 1`] = `
<div>
  <div
    class="col"
  >
    <div
      class="card"
    >
      <img
        class="card-img-top card-image"
        src="https://www.metaweather.com/static/img/weather/png/icon.png"
      />
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          Monday
        </div>
        <p
          class="card-text"
        >
          Min temp: 
          18
        </p>
        <p
          class="card-text"
        >
          Max temp: 
          20
        </p>
      </div>
    </div>
  </div>
</div>
`;
